watertap/unit_models/zero_order/tests/test_conventional_activated_sludge_zo.py

```diff
@@ -0,0 +1,256 @@
+    def test_conservation(self, model):
+        for j in model.fs.params.component_list:
+            assert 1e-6 >= abs(value(
+                model.fs.unit.inlet.flow_mass_comp[0, j] -
+                model.fs.unit.treated.flow_mass_comp[0, j] -
+                model.fs.unit.byproduct.flow_mass_comp[0, j]))
+
+    @pytest.mark.component
+    def test_report(self, model):
+        stream = StringIO()
+
+        model.fs.unit.report(ostream=stream)
+
+        output = """
+====================================================================================
+Unit : fs.unit                                                             Time: 0.0
+------------------------------------------------------------------------------------
+    Unit Performance
+
+    Variables: 
+
+    Key                              : Value   : Fixed : Bounds
+                  Electricity Demand :  6.5520 : False : (0, None)
+               Electricity Intensity : 0.14000 :  True : (None, None)
+                Solute Removal [foo] :  0.0000 :  True : (0, None)
+                Solute Removal [tss] : 0.50000 :  True : (0, None)
+    Solute Removal [viruses_enteric] : 0.99000 :  True : (0, None)
+                      Water Recovery : 0.99990 :  True : (1e-08, 1.0000001)
+
+------------------------------------------------------------------------------------
+    Stream Table
+                                         Inlet   Treated  Byproduct
+    Volumetric Flowrate                0.013000 0.011509  0.0014910
+    Mass Concentration H2O               769.23   868.80    0.67069
+    Mass Concentration viruses_enteric   76.923  0.86889     663.98
+    Mass Concentration tss               76.923   43.444     335.35
+    Mass Concentration foo               76.923   86.889 5.3655e-07
+====================================================================================
+"""
+
+        assert output in stream.getvalue()
+
+db = Database()
+params = db._get_technology("conventional_activated_sludge")
+
+
+class Test_CASZOsubtype:
+    @pytest.fixture(scope="class")
+    def model(self):
+        m = ConcreteModel()
+
+        m.fs = FlowsheetBlock(default={"dynamic": False})
+        m.fs.params = WaterParameterBlock(
+            default={"solute_list": ["viruses_enteric"]})
+
+        m.fs.unit = CASZO(default={
+            "property_package": m.fs.params,
+            "database": db})
+
+        return m
+
+    @pytest.mark.parametrize("subtype", [params.keys()])
+    @pytest.mark.component
+    def test_load_parameters(self, model, subtype):
+        model.fs.unit.config.process_subtype = subtype
+        data = db.get_unit_operation_parameters("conventional_activated_sludge", subtype=subtype)
+
+        model.fs.unit.load_parameters_from_database()
+
+        for (t, j), v in model.fs.unit.removal_frac_mass_solute.items():
+            assert v.fixed
+            assert v.value == data["removal_frac_mass_solute"][j]["value"]
+
+
+def test_costing():
```

Comment:
Maybe also test the subtype for `test_costing`.

Reply:
OK, done!

Reply:
I see you added the subtype test for costing the `test_secondary_treatment_wwtp_zo.py`, but here I denote the `test_conventional_activated_sludge_zo.py` since the model has two subtypes.

Reply:
Good catch. I added it in.